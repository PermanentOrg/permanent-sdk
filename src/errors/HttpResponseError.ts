export class HttpResponseError extends Error {
  public readonly statusCode: number;

  public constructor(statusCode: number, message: string) {
    super(message);
    this.statusCode = statusCode;
  }
}
