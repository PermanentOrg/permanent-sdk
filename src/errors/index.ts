export * from './HttpResponseError';
export * from './ValidationError';
