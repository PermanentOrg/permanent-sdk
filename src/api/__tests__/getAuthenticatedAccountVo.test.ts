import path from "node:path";
import nock from "nock";
import { getAuthenticatedAccountVo } from "..";
import { ValidationError } from "../../errors";

describe("getAuthenticatedAccountVo", () => {
	it("should return the account VO", async () => {
		nock("https://permanent.local")
			.get("/api/account/getAuthenticatedAccount")
			.replyWithFile(
				200,
				path.join(
					__dirname,
					"fixtures",
					"getAuthenticatedAccountVo",
					"accountVo.json",
				),
				{
					"Content-Type": "application/json",
				},
			);

		const accountVo = await getAuthenticatedAccountVo({
			bearerToken: "12345",
			baseUrl: "https://permanent.local/api",
		});

		expect(accountVo).toMatchSnapshot();
	});

	it("should error when provided an invalid response", async () => {
		nock("https://permanent.local")
			.get("/api/account/getAuthenticatedAccount")
			.reply(200, '["Not at all an AccountVo"]', {
				"Content-Type": "application/json",
			});

		await expect(
			getAuthenticatedAccountVo({
				bearerToken: "12345",
				baseUrl: "https://permanent.local/api",
			}),
		).rejects.toBeInstanceOf(ValidationError);
	});
});
