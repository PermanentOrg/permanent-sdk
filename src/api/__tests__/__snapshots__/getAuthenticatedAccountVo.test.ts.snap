// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAuthenticatedAccountVo should return the account VO 1`] = `
Object {
  "accessRole": null,
  "accountId": 2,
  "address": null,
  "address2": null,
  "agreed": null,
  "betaParticipant": null,
  "changePrimaryEmail": null,
  "changePrimaryPhone": null,
  "city": null,
  "country": null,
  "createdDT": "2023-03-09T17:33:09",
  "defaultArchiveCreated": 1,
  "defaultArchiveId": 1,
  "emailArray": null,
  "emailStatus": "status.auth.unverified",
  "fileLeft": 100006,
  "fileTotal": null,
  "fullName": "Blarg Blarg",
  "inviteCode": null,
  "keepLoggedIn": null,
  "notificationPreferences": "{\\"textPreference\\": {\\"apps\\": {\\"confirmations\\": 1}, \\"share\\": {\\"requests\\": 1, \\"activities\\": 1, \\"confirmations\\": 1}, \\"account\\": {\\"confirmations\\": 1, \\"recommendations\\": 1}, \\"archive\\": {\\"requests\\": 1, \\"confirmations\\": 1}, \\"relationships\\": {\\"requests\\": 1, \\"confirmations\\": 1}}, \\"emailPreference\\": {\\"apps\\": {\\"confirmations\\": 1}, \\"share\\": {\\"requests\\": 1, \\"activities\\": 1, \\"confirmations\\": 1}, \\"account\\": {\\"confirmations\\": 1, \\"recommendations\\": 1}, \\"archive\\": {\\"requests\\": 1, \\"confirmations\\": 1}, \\"relationships\\": {\\"requests\\": 1, \\"confirmations\\": 1}}, \\"inAppPreference\\": {\\"apps\\": {\\"confirmations\\": 1}, \\"share\\": {\\"requests\\": 1, \\"activities\\": 1, \\"confirmations\\": 1}, \\"account\\": {\\"confirmations\\": 1, \\"recommendations\\": 1}, \\"archive\\": {\\"requests\\": 1, \\"confirmations\\": 1}, \\"relationships\\": {\\"requests\\": 1, \\"confirmations\\": 1}}}",
  "optIn": null,
  "phoneStatus": "status.auth.none",
  "primaryEmail": "blarg@permanent.org",
  "primaryPhone": null,
  "spaceLeft": 68341120,
  "spaceTotal": 104857600,
  "state": null,
  "status": "status.auth.ok",
  "subject": "6a717417-7717-4767-b0be-1afb6ca8dbd5",
  "type": "type.account.standard",
  "updatedDT": "2023-03-09T17:33:13",
  "zip": null,
}
`;
